readthedocs/core/signals.py

```diff
@@ -79,12 +80,16 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
 def delete_projects_and_organizations(sender, instance, *args, **kwargs):
     # Here we count the owner list from the projects that the user own
     # Then exclude the projects where there are more than one owner
-    projects = instance.projects.all().annotate(num_users=Count('users')).exclude(num_users__gt=1)
+    # Add annotate before filter
+    # https://bit.ly/2Nne6ZJ
+    projects = (Project.objects.annotate(num_users=Count('users')).filter(users=instance.id)
+                                                                  .exclude(num_users__gt=1))
 
     # Here we count the users list from the organization that the user belong
     # Then exclude the organizations where there are more than one user
-    oauth_organizations = (instance.oauth_organizations.annotate(num_users=Count('users'))
-                                                       .exclude(num_users__gt=1))
+    oauth_organizations = (RemoteOrganization.objects.annotate(num_users=Count('users'))
+                                                     .filter(users=instance.id)
+                                                     .exclude(num_users__gt=1))
```

Comment:
These changes should conform to our styleguide by running `pre-commit` on the file

Reply:
I think it does not break PEP8 styleguide. Can you explain how it would change?

Reply:
We _have_ the style guide _inside_ pre-commit p:

Reply:
Yup, no explicit styleguide. it's all enforced by pre-commit via yapf and friends. In this specific case, we opt for no right side alignment of variables and arguments. Both are pep8 compliant, but right side alignment is too unreadable and annoying to maintain.

Reply:
Info on pre-commit procedure is here:
https://docs.readthedocs.io/en/latest/contribute.html#contributing-to-development